tape/src/main/java/com/squareup/tape/QueueFile.java

```diff
@@ -211,9 +211,29 @@ private Element readElement(int position) throws IOException {
     if (position == 0) return Element.NULL;
     ringRead(position, buffer, 0, Element.HEADER_LENGTH);
     int length = readInt(buffer, 0);
+    checkForElementLengthCorruption(length);
     return new Element(position, length);
   }
 
+  /**
+   * If a single element in the queue has a size greater than the length of the content as recorded
+   * in the header, we have a corruption problem. We also check against maxElementSize as no element
+   * should have been inserted with a size greater than that - so if the element's header reports as
+   * such, then again there is corruption.
+   *
+   * @param elementLength - Length of the element we are checking.
+   * @throws IOException - When there is a file read problem or when there is corruption.
+   */
+  private void checkForElementLengthCorruption(int elementLength) throws IOException {
+    int fileContentLength = fileLength - QueueFile.HEADER_LENGTH;
+    if (elementLength > fileContentLength || elementLength > maxElementSize) {
+      throw new IOException("Possible corruption: Trying to read a Tape element "
```

Comment:
Why not reuse FileException here? Tape handling code generally expects FileException no?

https://github.com/square/tape/blob/0e8bfc8dcb42b4ee0ddd5c0bd06063952bd59bd9/tape/src/main/java/com/squareup/tape/FileException.java

Reply:
only for the read/write operations. For the creation of the queues they are all marked as throwing IOException